src/trust_manager.rs

```diff
@@ -0,0 +1,288 @@
+use crate::common::{Key, Value};
+use crate::storage_manager::{StorageManagerError, StorageManagerInterface};
+
+/// Internal prefix for trusted did keys.
+const KEY_PREFIX: &str = "TrustedDIDs.";
+
+#[derive(thiserror::Error, Debug, uniffi::Error)]
+pub enum TrustManagerError {
+    #[error("An unexpected foreign callback error occurred: {0}")]
+    UnexpectedUniFFICallbackError(String),
+    #[error(transparent)]
+    Storage(#[from] StorageManagerError),
+    #[error("The DID key cannot be added because it is blocked, key: {0}")]
+    DIDBlocked(String),
+}
+
+// Handle unexpected errors when calling a foreign callback
+impl From<uniffi::UnexpectedUniFFICallbackError> for TrustManagerError {
+    fn from(value: uniffi::UnexpectedUniFFICallbackError) -> Self {
+        TrustManagerError::UnexpectedUniFFICallbackError(value.reason)
+    }
+}
+
+/// TrustManager is responsible for managing trusted DIDs for the wallet.
+///
+/// Use the [TrustManager::new] method to create a new instance of the trust manager.
+///
+/// The trust manager does not store a cached state of the trusted dids,
+/// but instead accesses and modifies the trusted dids in the storage manager directly.
+///
+/// In the future, this might change in favor of faster reads.
+#[derive(Debug)]
+pub struct TrustManager;
```

Comment:
This really isn't a bad idea, but there's a huge **_but_**. We can't just add a DID to this manager and assume every transaction we do with that DID can be trusted. For example, we might trust `did:web:suspisciousonlineshop.com` for verifying a proof-of-age credential, but we probably don't trust it with PII heavy credentials.

We might want to build something in this shape in the future, but it would need an extensive "capabilities" interface that allows a wallet to define what kinds of transactions a DID is trusted for. Whether or not a wallet trusts a verifier has a lot to do with the context of the transaction. For those reasons, I think this kind of interface is too much work and should be out of scope for this task. Instead, we should just expose an interface for an implementor to decide whether or not to trust a specific DID at transaction time.

Reply:
Tracking this conversation in #33